File: AUGUST-2025/14-08-2025/p2264.ex
defmodule Solution do
  @spec largest_good_integer(num :: String.t) :: String.t
  def largest_good_integer(num) do
    ~r/(?<triplet>(.)\2\2)\2*/
    |> Regex.scan(num, capture: [:triplet])
    |> Enum.max(fn -> [""] end)
    |> hd()
  end
end

File: AUGUST-2025/14-08-2025/p2264.js
/**
 * @param {string} num
 * @return {string}
 */
var largestGoodInteger = function(num) {
    if (num.length < 3) return ''

    let l = 0;
    let max = -Infinity;
    let rez;    
    for (let r = 1; r < num.length; r ++){
        if (num[l] !== num[r]) l = r
        if (r - l + 1 === 3) {
            max = Math.max(max, +num[r]);
            rez = `${max}${max}${max}`;
        }
    }

    return max !== -Infinity ? rez : ''
};